ReactWindows/ReactNative/Views/TextInput/ReactTextBox.cs

```diff
@@ -0,0 +1,87 @@
+﻿using ReactNative.UIManager;
+using System.Threading;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace ReactNative.Views.TextInput
+{
+    class ReactTextBox : TextBox, ILayoutManager
+    {
+        private int _eventCount;
+        private double _lastWidth;
+        private double _lastHeight;
+
+        public ReactTextBox()
+        {
+            LayoutUpdated += OnLayoutUpdated;
+        }
+
+        public int CurrentEventCount
+        {
+            get
+            {
+                return _eventCount;
+            }
+        }
+
+        public bool ClearTextOnFocus
+        {
+            get;
+            set;
+        }
+
+        public bool SelectTextOnFocus
+        {
+            get;
+            set;
+        }
+
+        public int IncrementEventCount()
+        {
+            return Interlocked.Increment(ref _eventCount);
+        }
+
+        public void UpdateLayout(int x, int y, int width, int height)
+        {
+            Canvas.SetLeft(this, x);
+            Canvas.SetTop(this, y);
+            Width = width;
+        }
+
+        protected override void OnGotFocus(RoutedEventArgs e)
+        {
+            if (ClearTextOnFocus)
+            {
+                Text = "";
+            }
+
+            if (SelectTextOnFocus)
+            {
+                SelectionStart = 0;
+                SelectionLength = Text.Length;
+            }
+        }
+
+        private void OnLayoutUpdated(object sender, object e)
```

Comment:
The LayoutUpdated event delegate type is EventHandler<System.Object>.

---
In reply to: [56671657](https://github.com/CatalystCode/react-native/pull/273#discussion_r56671657) [](ancestors = 56671657)